### PR TITLE
fix(Contacts): Add self id for identity verification requests

### LINKED ISSUE #8277: Responding to contact verification requests is broken
Tested this a bunch of times with @caybro now. 
Coming contacts is fine, however responding to a verification request sent by the other account doesn't seem to do anything.

UI-wise, it looks like the response was sent. However, looking at the logs, we see that it couldn't find the verification request to respond to in the first place:

```sh
DBG 2022-11-16 11:37:04.038+01:00 NewBE_callPrivateRPC                       topics="rpc" tid=39147 file=core.nim:20 rpc_method=wakuext_acceptContactVerificationRequest
ERR 2022-11-16 11:37:04.040+01:00 rpc response error                         topics="rpc" tid=39147 file=core.nim:29 err="\nstatus-go error [methodName:wakuext_acceptContactVerificationRequest, code:-32000, message:verification request not found ]\n"
ERR 2022-11-16 11:37:04.040+01:00 error doing rpc request                    topics="rpc" tid=39147 file=core.nim:33 methodName=wakuext_acceptContactVerificationRequest exception="\nstatus-go error [methodName:wakuext_acceptContactVerificationRequest, code:-32000, message:verification request not found ]\n"
ERR 2022-11-16 11:37:04.040+01:00 error accepting contact verification request topics="contacts-service" tid=39147 file=service.nim:693 msg="\nstatus-go error [methodName:wakuext_acceptContactVerificationRequest, code:-32000, message:verification request not found ]\n"
```

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #8277

### What does the PR do

Fixed:
- [x] Accept incoming verification request
- [x] Decline incoming verification request
- [x] Cancel outgoing verification request (https://github.com/status-im/status-desktop/issues/8437)
- [x] Verify trusted outgoing verification request
- [x] Verify untrustworthy outgoing verification request

Fix accepting and declining identity verification requests

### Affected areas

Contacts

### Screenshot of functionality (including design for comparison)
Screenshot 2022-11-23 at 17 14 51
Screenshot 2022-11-23 at 17 14 29
Screenshot 2022-11-23 at 17 14 20
Screenshot 2022-11-23 at 17 13 22
Screenshot 2022-11-23 at 16 26 59

### CODE REVIEW

**Comment by MishkaRogachev** [comment]
@jrainville @PascalPrecht accept/decline identity verification request were not working because we used public key instead of request.Id.  Where are two ways:
1) Switch on request.id everywhere, including QML. That's more solid but needs more time.
2) Use request.id only on service and status-go layers. That's the way how contact requests works nowadays. This solution is much faster but it "smells"

**Comment by jrainville** [comment]
@MishkaRogachev IMO, solution 2 sounds fine. The front-end doesn't necessarily need to know about request id for now. As long as the backend can do the operation easily, it's ok.
If you want, you can add a ticket to refactor that in the future, but it might never get done :sweat_smile:

**Comment by PascalPrecht** [comment]
I agree with @jrainville and think this is a pragmatic solution to get the non-working feature functional.
Great find @MishkaRogachev !

**Review by jrainville — APPROVED** [review]

File: src/app_service/service/contacts/service.nim [hunk 1]
```diff
@@ -560,34 +560,44 @@ QtObject:
       TrustArgs(publicKey: publicKey, isUntrustworthy: true))
 
   proc verifiedTrusted*(self: Service, publicKey: string) =
-    let response = status_contacts.verifiedTrusted(publicKey)
-    if(not response.error.isNil):
-      let msg = response.error.message
-      error "error confirming identity ", msg
-      return
+    try:
+      var response = status_contacts.getVerificationRequestSentTo(publicKey)
```
Comment (endulab) [inline src/app_service/service/contacts/service.nim:564]: this may return error

File: src/app_service/service/contacts/service.nim [hunk 2]
```diff
@@ -560,34 +560,44 @@ QtObject:
...
+        raise newException(RpcException, msg)
 
-    self.events.emit(SIGNAL_CONTACT_TRUSTED,
-      TrustArgs(publicKey: publicKey, isUntrustworthy: false))
-    self.events.emit(SIGNAL_CONTACT_VERIFIED, ContactArgs(contactId: publicKey))
+      if self.contacts.hasKey(publicKey):
+        self.contacts[publicKey].trustStatus = TrustStatus.Trusted
+        self.contacts[publicKey].verificationStatus = VerificationStatus.Trusted
+
+      self.events.emit(SIGNAL_CONTACT_TRUSTED,
```
Comment (endulab) [inline src/app_service/service/contacts/service.nim:576]: signals should be sent in "if" clause ?

File: src/app_service/service/contacts/service.nim [hunk 3]
```diff
@@ -560,34 +560,44 @@ QtObject:
...
+      error "error verified trusted request", msg=e.msg
 
   proc verifiedUntrustworthy*(self: Service, publicKey: string) =
-    let response = status_contacts.verifiedUntrustworthy(publicKey)
-    if(not response.error.isNil):
-      let msg = response.error.message
-      error "error confirming identity ", msg
-      return
+    try:
+      var response = status_contacts.getVerificationRequestSentTo(publicKey)
```
Comment (endulab) [inline src/app_service/service/contacts/service.nim:584]: this may return error

File: src/app_service/service/contacts/service.nim [hunk 4]
```diff
@@ -560,34 +560,44 @@ QtObject:
...
+        raise newException(RpcException, msg)
 
-    self.events.emit(SIGNAL_CONTACT_UNTRUSTWORTHY,
-      TrustArgs(publicKey: publicKey, isUntrustworthy: true))
-    self.events.emit(SIGNAL_CONTACT_VERIFIED, ContactArgs(contactId: publicKey))
+      if self.contacts.hasKey(publicKey):
+        self.contacts[publicKey].trustStatus = TrustStatus.Untrustworthy
+        self.contacts[publicKey].verificationStatus = VerificationStatus.Verified
+
+      self.events.emit(SIGNAL_CONTACT_UNTRUSTWORTHY,
```
Comment (endulab) [inline src/app_service/service/contacts/service.nim:596]: signals should be sent in if clause

**Review by endulab — CHANGES_REQUESTED** [review]

**Review by jrainville — APPROVED** [review]

**Review by endulab — APPROVED** [review]

**Pull request merged**